lib/collection/src/shards/replica_set/update.rs

```diff
@@ -424,9 +424,12 @@ impl ShardReplicaSet {
             Some(ReplicaState::Partial) => true,
             Some(ReplicaState::Initializing) => true,
             Some(ReplicaState::Listener) => true,
-            // We must not send updates to replicas in recovery state.
-            // If we do we might create gaps in WAL clock tags.
-            Some(ReplicaState::Recovery | ReplicaState::PartialSnapshot) => false,
+            // Recovery: keep sending updates to prevent a data race
+            // The replica on the peer may still be active for some time if its consensus is slow.
+            // The peer may respond to read requests until it switches to recovery state too. We
+            // must keep sending updates to prevent those reads being stale.
+            // See: <https://github.com/qdrant/qdrant/pull/5298>
+            Some(ReplicaState::Recovery | ReplicaState::PartialSnapshot) => true,
```

Comment:
Do you think we should revert this part as well?

Reply:
I'm indecisive thus far, I left a comment about this in my notes.

Based on what we've discussed before I'd be happy to keep it at false.

Reply:
Kept this at false - we still do not send updates to replicas in recovery state.